OSDOCS-6528: Created documentation for creating a shared VPC (#63793)
openshift/openshift-docs: modules/rosa-sharing-vpc-creation-and-sharing.adoc

// Module included in the following assemblies:
//
// * networking/rosa-shared-vpc-config.adoc

:_content-type: PROCEDURE
[id="rosa-sharing-vpc-creation-and-sharing_{context}"]
= Configuring a VPC to share within your AWS organization

You can share your configured VPC with another AWS user account if that account is within your current AWS organization.

.Procedure

> . On your VPC-owning AWS account ID, you must create your VPC to your desired specifications in the link:https://us-east-1.console.aws.amazon.com/vpc/[VPC section of the AWS console]. 

Reviewer: This sentence confuses me. If the AWS account already owns the VPC (per the first phrase), why do you need to create the VPC?

. In the link:https://us-east-1.console.aws.amazon.com/iamv2/[Identity and Access Management (IAM) section of the AWS console], create a new custom trust policy role for the shared VPC permissions. This role should have the following permissions:
  * A trust policy to assume roles:
+
[source,terminal]
----
{
	"Version": "2012-10-17",
	"Statement": [
		{
			"Sid": "Statement1",
			"Effect": "Allow",

> 			"Principal": {}, <1>

Reviewer 2: AWS console validates the principal, and the user won't be able to create a policy with an empty principal.
The workaround is to set the root user as a placeholder, as described here https://source.redhat.com/groups/public/ocm_team/ocm_blog/creating_a_cluster_in_a_shared_vpc_support_preexisting_route53_for_byo_vpc_clusters

			"Action": "sts:AssumeRole"
		}
	]
}
----
+
--
<1> The principals are added later in this process once the cluster-creating AWS account user has created these roles.
--
    * `ResourceGroupandTagEditorFullAccess` permissions policy
    * `Route53minimalPermissions` permissions policy
+
After you create this IAM role, you must give the created role's ARN to the cluster creator.

> . In the link:https://us-east-1.console.aws.amazon.com/ram/[Resource Access Manager of the AWS console], create a Resource Share that includes your VPC's private and public subnets and the cluster-creating AWS account ID.

Reviewer 3: ```suggestion
. In the link:https://us-east-1.console.aws.amazon.com/ram/[Resource Access Manager of the AWS console], create a Resource Share that shares your VPC's private and public subnets with the cluster-creating AWS account ID.
```

> After creating this Resource Share, the cluster-creating AWS account user must create a DNS domain and operator roles to continue configuration.

Reviewer: I think it might be clearer if the two separate users were always referred to in a similar manner. For example, in this sentence you write "cluster-creating AWS account user" while in the previous sentence it reads "the cluster creator".

> After creating this Resource Share, the cluster-creating AWS account user must create a DNS domain and operator roles to continue configuration.

Reviewer: I think "operator" needs to be capitalized.

// Module included in the following assemblies:
//
// * networking/rosa-shared-vpc-config.adoc

:_content-type: PROCEDURE
[id="rosa-sharing-vpc-creation-and-sharing_{context}"]
= Configuring a VPC to share within your AWS organization

You can share your configured VPC with another AWS user account if that account is within your current AWS organization.

.Procedure

> . On one AWS account ID, create a VPC to your desired specifications in the link:https://us-east-1.console.aws.amazon.com/vpc/[VPC section of the AWS console]. This AWS account is now your VPC-owning AWS account. 

Author: @wgordon17, on line 13, should:

>  From your centrally managed VPC AWS account, create or modify a
be 
>  From your centrally managed AWS account, create or modify a

// Module included in the following assemblies:
//
// * networking/rosa-shared-vpc-config.adoc

:_content-type: PROCEDURE
[id="rosa-sharing-vpc-creation-and-sharing_{context}"]
= Configuring a VPC to share within your AWS organization

You can share subnets within a configured VPC with another AWS user account if that account is within your current AWS organization.

.Procedure

> . From the AWS account that centrally manages your VPC, create or modify a VPC to your desired specifications in the link:https://us-east-1.console.aws.amazon.com/vpc/[VPC section of the AWS console]. This AWS account will be the *VPC-owning AWS account*. 

Reviewer 4: ```suggestion
. From the AWS account that centrally manages your VPC, create or modify a VPC to your specifications in the link:https://us-east-1.console.aws.amazon.com/vpc/[VPC section of the AWS console]. This AWS account will be the *VPC-owning AWS account*. 
```
- IBM Style advises against the use of "desired" in technical information.
- Double check the highlighting of "VPC-owning AWS account". AFAIK, we mostly use bold highlighting for UI elements.

> . In the link:https://us-east-1.console.aws.amazon.com/iamv2/[Identity and Access Management (IAM) section of the AWS console], create a new custom trust policy role for the shared VPC permissions. This role should have the following permissions:

Reviewer 4: ```suggestion
. In the link:https://us-east-1.console.aws.amazon.com/iamv2/[Identity and Access Management (IAM) section of the AWS console], create a custom trust policy role for the shared VPC permissions. Ensure that the role has the following permissions:
```
- Per IBM Style, "create a new" can be simplified to "create a".
- Per IBM Style, avoid modal verbs (would, should, could) if possible.

>   * A trust policy to assume roles:

Reviewer 4: Will the intended audience understand where to add or modify the trust policy? Consider whether it might be helpful to tell the user how they can get a trust policy.

+
[source,terminal]
----
{
  "Version": "2012-10-17",
  "Statement": [
    {
	  "Sid": "Statement1",
	  "Effect": "Allow",
	  "Principal": {
	  	"AWS": "arn:aws:iam::<Account-ID>:root"
	  }, <1>
	  "Action": "sts:AssumeRole"
	}
  ]
}
----
+
--

> <1> The below principals will be added later in this process once the *cluster-creating AWS account* user has created these roles. On creation, you should create a root user placeholder using the *cluster-creator's AWS Account* ID as `arn:aws:iam::{Account}:root`.

Reviewer 4: ```suggestion
<1> The following principles are added later in this process after the *cluster-creating AWS account* user has created these roles. On creation, you must create a root user placeholder by using the *cluster-creator's AWS Account* ID as `arn:aws:iam::{Account}:root`.
```
- Check the use of bold highlighting here.

--
    * `ResourceGroupandTagEditorFullAccess` permissions policy

>     * `Route53minimalPermissions` permissions policy

Reviewer 4: ```suggestion
    * The `Route53minimalPermissions` permissions policy
```

+

> After you create this IAM role, provide the created role's ARN to the cluster creator.

Reviewer 4: Does ARN need to be spelled out, or will most users know what it means?

> . In the link:https://us-east-1.console.aws.amazon.com/ram/[Resource Access Manager of the AWS console], create a Resource Share that shares the previously created public and/or private subnets to the *cluster-creating AWS account* ID.

Reviewer: I don't think we are supposed to use "and/or" because it causes translation errors.

> . In the link:https://us-east-1.console.aws.amazon.com/ram/[Resource Access Manager of the AWS console], create a Resource Share that shares the previously created public and private subnets to the *cluster-creating AWS account* ID.

Reviewer 4: On lines 42 and 44, check whether bold highlighting is correct for the ID names

> After creating this Resource Share, notify the *cluster-creating AWS account* user to reserve an _openshiftapps.com_ DNS domain and create Operator roles to continue configuration.

Reviewer 4: ```suggestion
After you create the Resource Share, notify the *cluster-creating AWS account* user to reserve an _openshiftapps.com_ DNS domain and create Operator roles to continue configuration.
```
